### ISSUE TITLE
echo_server.go receives only null bytes, echo_client.go segfaults upon receiving the echoed package

### ISSUE THREAD

**Issue by sauerbraten** [deft-code/go-enet#1]
I tried running the server and client, but upon sending something from the client, for example "1", the server received [0] and the cient segfaulted upon receiving the server's response. I tried all this with enet 1.3.9. Maybe the protocol has changed or something?

**Comment by deft-code** [comment]
I ran into the null byte issue myself during development. I was never able to pin it down. I suspect it might have something to do with cgo.

**Comment by sauerbraten** [comment]
Yeah I couldn't pin point it either. I even added prints in the enet library, but couldn't figure it out. Wireshark also showed the first eight byte nulled. Writing the complete server in C, then calling it from Go works, though.
